test/controllers/EtmpCheckControllerTest.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import models.EtmpRegistrationDetails
+import org.mockito.Matchers
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import play.api.libs.json.Json
+import play.api.mvc.ControllerComponents
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.EtmpRegimeService
+import utils.{BaseSpec, AwrsTestJson}
+
+import scala.concurrent.Future
+
+class EtmpCheckControllerTest extends BaseSpec with MockitoSugar {
+
+  val mockEtmpRegimeService: EtmpRegimeService = mock[EtmpRegimeService]
+  val cc: ControllerComponents = app.injector.instanceOf[ControllerComponents]
+
+  object TestEtmpCheckController extends EtmpCheckController(cc, mockEtmpRegimeService)
+
+  "checkEtmp" should {
+    "return an OK" when {
+      "there is a regime model and there is ETMP registration details for an organisation" in {
+        val etmpRegistrationDetails = EtmpRegistrationDetails(
+          None,
+          "sapNumber",
+          "safeId",
+          None,
+          "regRef",
+          None
+        )
+
+          when(mockEtmpRegimeService.checkETMPApi(Matchers.any(), Matchers.any(), Matchers.any())(Matchers.any(), Matchers.any()))
+            .thenReturn(Future.successful(Some(etmpRegistrationDetails)))
+
+        val result = TestEtmpCheckController.checkEtmp().apply(FakeRequest().withJsonBody(Json.parse(etmpCheckOrganisationString)))
+        status(result) shouldBe OK
+      }
+
+      "there is a regime model and there is ETMP registration details for an individual" in {
```

Comment:
RED - The descriptions for these tests sound like they should be passing? They have the same descriptions as the happy path tests.

OK scenario - "There is a regime model and there is ETMP registration details for an individual"
NoContent scenario = "there is a regime model and there is ETMP registration details for an individual"

It looks like you're passing invalid details for the second one but that should be clear.